### PR TITLE
Add SuppressLint("ClickableViewAccessibility") to OnTouchListener's

### LINKED ISSUE #772: Apply @SuppressLint attribute to generated handler methods when necessary
I have my `lint.xml` set up so that `ClickableViewAccessibility` raises an `error`. 

The `@OnTouch` binding does not have a way to suppress that warning when I am making sure to call `performClick` when necessary in my annotated method.

I recommend either picking up `@SupressLint` annotations on bound methods and applying them to the generated listener, or adding an optional string parameter to the `@On[Event]` annotations themselves for this situation.

### PR DESCRIPTION
**Pull request by ChristianBecker**
Closes #772

### CODE REVIEW

File: butterknife-compiler/src/main/java/butterknife/compiler/BindingSet.java [hunk 1]
```diff
@@ -542,6 +551,16 @@ private boolean hasMethodBindings() {
     return false;
   }
 
+  private boolean hasOnTouchMethodBindings() {
+    for (ViewBinding bindings : viewBindings) {
+      if (bindings.getMethodBindings()
+          .containsKey(OnTouch.class.getAnnotation(ListenerClass.class))) {
```
Comment (ChristianBecker) [inline butterknife-compiler/src/main/java/butterknife/compiler/BindingSet.java]: I'm open to suggestions if you are not happy with the explicit dependency on the OnTouch annotation

File: butterknife/src/test/java/butterknife/OnTouchTest.java [hunk 2]
```diff
@@ -33,6 +34,7 @@
         + "  private Test target;\n"
         + "  private View view1;\n"
         + "  @UiThread\n"
+        + "  @SuppressLint(\"ClickableViewAccessibility\")\n"
```
Comment (ChristianBecker) [inline butterknife/src/test/java/butterknife/OnTouchTest.java]: If you want I can move this annotation directly to the `onTouch(..)` implementation to reduce the scope.

Comment (JakeWharton) [inline butterknife/src/test/java/butterknife/OnTouchTest.java]: Yes I would prefer this, if it's not too difficult.

Comment (ChristianBecker) [inline butterknife/src/test/java/butterknife/OnTouchTest.java]: Unfortunately, it's not that easy without extending `ListenerMethod` or without including another hard-coded check for `OnTouch` in `BindingSet#addMethodBindings(..)`. What do you think?

Comment (JakeWharton) [inline butterknife/src/test/java/butterknife/OnTouchTest.java]: Ok. Not a big deal.

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**